lib/sqlalchemy/sql/elements.py

```diff
@@ -3742,6 +3781,21 @@ def _literal_and_labels_as_label_reference(element):
         return _literal_as_text(element)
 
 
+def _bool_expression_literal_as_text(element):
+    """Converts a boolean expression into text, but warn if None is passed in, to avoid NULL boolean expressions"""
+    if element is None:
+        if NONE_CLAUSE_HANDLING in ("warning", "error") or NONE_CLAUSE_HANDLING.startswith("log:"):
+            if NONE_CLAUSE_HANDLING == "warning":
+                warnings.warn("Use of None in expression", NoneClauseWarning, 2)
+            elif NONE_CLAUSE_HANDLING == "error":
+                raise NoneClauseError("Use of None in expression")
+            else:
+                level = getattr(logging, NONE_CLAUSE_HANDLING.replace("log:", "", 1).upper(), logging.ERROR)
+                logging.log(level, "Use of None in expression")
+        return TextClause("")
```

Comment:
Nope, it should always return that if `element is None`; unless it's raising an error, in which case it won't reach here.